src/bindings/fluid_ladspa.c

```diff
@@ -160,6 +162,43 @@ void delete_fluid_ladspa_fx(fluid_ladspa_fx_t *fx)
 };
 
 /**
+ * Set the sample rate of the LADSPA effects.
+ *
+ * Resets the LADSPA effects if the sample rate is different from the
+ * previous sample rate.
+ *
+ * @param fx LADSPA fx instance
+ * @param sample_rate new sample rate
+ */
+void fluid_ladspa_set_sample_rate(fluid_ladspa_fx_t *fx, fluid_synth_t *synth)
+{
+    unsigned long new_sample_rate = (unsigned long)synth->sample_rate;
```

Comment:
`synth->sample_rate` is mutex protected, potentially causing a dirty read here. Prefer handing `new_sample_rate` directly into `fluid_ladspa_set_sample_rate()` as extra param.

Also consider doing this:

`(unsigned long)(synth->sample_rate + 0.5) /* to minimize overall casting error */`